src/System.IO.FileSystem/tests/File/Create_str.cs

```diff
@@ -0,0 +1,231 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public class File_Create_str : FileSystemTest
+    {
+        #region Utilities
+
+        public virtual FileStream Create(string path)
+        {
+            return File.Create(path);
+        }
+
+        #endregion
+
+        #region UniversalTests
+
+        [Fact]
+        public void NullPath()
+        {
+            Assert.Throws<ArgumentNullException>(() => Create(null));
+        }
+
+        [Fact]
+        public void EmptyPath()
+        {
+            Assert.Throws<ArgumentException>(() => Create(string.Empty));
+        }
+
+        [Fact]
+        public void CreateCurrentDirectory()
+        {
+            Assert.Throws<UnauthorizedAccessException>(() => Create("."));
+        }
+
+        [Fact]
+        public void ValidCreation()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName());
+            using (FileStream stream = Create(testFile))
+            {
+                Assert.True(File.Exists(testFile));
+                Assert.Equal(0, stream.Length);
+                Assert.Equal(0, stream.Position);
+            }
+        }
+
+        [Fact]
+        public void CreateInParentDirectory()
+        {
+            string testFile = GetTestFileName();
+            using (FileStream stream = Create(Path.Combine(TestDirectory, "DoesntExists", "..", testFile)))
+            {
+                Assert.True(File.Exists(Path.Combine(TestDirectory, testFile)));
+            }
+        }
+
+        [Fact]
+        public void LegalSymbols()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName() + "!@#$%^&");
+            using (FileStream stream = Create(testFile))
+            {
+                Assert.True(File.Exists(testFile));
+            }
+        }
+
+        [Fact]
+        public void InvalidDirectory()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName(), GetTestFileName());
+            Assert.Throws<DirectoryNotFoundException>(() => Create(testFile));
+        }
+
+        [Fact]
+        public void ValidWhitespaceNames()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile1 = Path.Combine(testDir.FullName, GetTestFileName());
+            string testFile3 = Path.Combine(" ", testDir.FullName, " ", GetTestFileName());
+            using (Create(string.Format(" {0}", testFile1)))
+            using (Create(testFile3))
+            {
+                Assert.True(File.Exists(testFile1));
```

Comment:
Alright, I figured it out. My Unix build was outdated and didn't have the changes you made to Path. In the build I have, Path.Combine will just remove the extra whitespace from the path for both files, making the paths absolute and removing the directory with the one-space name.

There are other changes in this PR that will be altered by me updating my Unix build. This also explains why I reopened that issue - the changes weren't reflected yet to me. I'm going to change the way I test on Unix so that this doesn't happen again.

#2205 can likely be closed.